src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -1,3 +1,4 @@
+# TODO: Extract SCM-specific code to separate classes. This class is too big for GitHub and GitLab.
```

Comment:
I created [a card in the backlog](https://trello.com/c/E9jJl8Mb/2414-extract-scm-specific-code-from-triggercontrollerservicescmextractor-to-separate-classes) for this.